AngularJS/app.js

```diff
@@ -0,0 +1,214 @@
+angular.module("reportApp", ['main']);
+var app = angular.module("main", []);
+
+angular.module('reportApp')
+    .service('reportService', ReportService)
+    .component('reportList', {
+        templateUrl: 'reportList.html',
+        controller: ReportListController
+    })
+
+function ReportListController(reportService) {
+    //Lists all reports in tabular form, with edit and remove options against each report item.
+    var $ctrl = this;
+    var report_config = {
+        //config object defined in current controller's scope
+        //for the accessing constant variables.
+        current_report_item: [],
+        auth_token: "",
+        api_url: "http://localhost:8000/",
+        types: [{
+            keys: "Annual",
+            value: "AN"
+        }, {
+            keys: "Monthly",
+            value: "MO"
+        }, {
+            keys: "Weekly",
+            value: "WE"
+        }],
+        init: () => {
+            //hides edit form and populates dropdown for add form
+            $ctrl.editForm = true;
+            $ctrl.types = report_config.types;
+            $ctrl.report_type = $ctrl.types[0];
+        }
+    }
+    reportService.get_token(report_config)
+        //gets token from service to authenticate.
+        .then(set_token,
+            function(errorMessage) {
+                console.warn(errorMessage);
+            }
+        );
+    reportService.get_reports(report_config)
+        .then(list_reports,
+            function(errorMessage) {
+                console.warn(errorMessage);
+            }
+        );
+    $ctrl.editReportItem = function(report_item) {
+        //populates form fields with selected report attribute.
+        $ctrl.form = false;
```

Comment:
Use appropriate variable names
`form` --> `showAddForm`

or create object for views like
```
views = {
    addForm: false,
    editForm: true ...
}
```